apm-agent-plugins/apm-scheduled-annotation-plugin/src/test/java/co/elastic/apm/agent/spring/scheduled/TimerTaskInstrumentationTest.java

```diff
@@ -0,0 +1,53 @@
+package co.elastic.apm.agent.spring.scheduled;
+
+import co.elastic.apm.agent.AbstractInstrumentationTest;
+import org.junit.jupiter.api.Test;
+
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TimerTaskInstrumentationTest extends AbstractInstrumentationTest {
+
+    @Test
+    void testTimerTaskWithScheduleAtFixedRate() throws InterruptedException {
+        reporter.reset();
+        TestTimerTask timerTask = new TestTimerTask();
+        Timer timer = new Timer(true);
+        timer.scheduleAtFixedRate(timerTask, 0, 5);
+        Thread.sleep(100);
```

Comment:
Use `await()` or `awaitUntilAsserted()` instead of sleep.

Reply:
replaced